src/main/java/tools/MaplePacketCreator.java

```diff
@@ -8318,17 +8318,24 @@ public static void addCashItemInformation(final MaplePacketLittleEndianWriter mp
                 mplew.writeInt(transition);
                 return mplew.getPacket();
         }
-        
-        public static byte[] setNPCScriptable(Set<Pair<Integer, String>> scriptNpcDescriptions) {  // thanks to GabrielSin
+
+        /**
+         * Makes the NPCs provided set as scriptable, informing the client to search for js scripts for these NPCs even
+         * if they already have entries within the wz files.
+         *
+         * @param scriptableNpcIds Ids of npcs to enable scripts for.
+         * @return a packet which makes the npc's provided scriptable.
+         */
+        public static byte[] setNPCScriptable(List<Integer> scriptableNpcIds) {  // thanks to GabrielSin
                 MaplePacketLittleEndianWriter mplew = new MaplePacketLittleEndianWriter();
                 mplew.writeShort(SendOpcode.SET_NPC_SCRIPTABLE.getValue());
-                mplew.write(scriptNpcDescriptions.size());
-                for (Pair<Integer, String> p : scriptNpcDescriptions) {
-                        mplew.writeInt(p.getLeft());
-                        mplew.writeMapleAsciiString(p.getRight());
+                mplew.write(scriptableNpcIds.size());
+                scriptableNpcIds.forEach(id -> {
+                        mplew.writeInt(id);
+                        mplew.writeMapleAsciiString("NPC " + id); // The client needs a name for the npc, but it doesn't seem to do anything.
```

Comment:
I have no idea how the client treats this value, and I don't feel like testing it. LGTM.